rules/coding-style/src/Rector/Ternary/TernaryConditionVariableAssignmentRector.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Ternary;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Scalar\LNumber;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\Ternary\TernaryConditionVariableAssignmentRector\TernaryConditionVariableAssignmentRectorTest
+ */
+final class TernaryConditionVariableAssignmentRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Assign outcome of ternary condition to variable, where applicable', [
+            new CodeSample(
+                <<<'PHP'
+function ternary()
+{
+    true ? $a = 1 : $a = 0;
+}
+PHP
+                ,
+                <<<'PHP'
+function ternary()
+{
+    $a = true ? 1 : 0;
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Ternary::class];
+    }
+
+    /**
+     * @param Ternary $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $nodeIf = $node->if;
+        $nodeElse = $node->else;
+
+        if (! $nodeIf instanceof Assign || ! $nodeElse instanceof Assign) {
+            return null;
+        }
+
+        $nodeIfVar = $nodeIf->var;
+        $nodeElseVar = $nodeElse->var;
+
+        if (! $nodeIfVar instanceof Variable || ! $nodeElseVar instanceof Variable) {
+            return null;
+        }
+
+        if ($nodeIfVar->name !== $nodeElseVar->name) {
+            return null;
+        }
+
+        $previousNode = $node->getAttribute(AttributeKey::PREVIOUS_NODE);
+        if ($previousNode !== null) {
+            return null;
+        }
+
+        $variable = new Variable($nodeIfVar->name);
+
+        $nodeIfExpr = $nodeIf->expr;
+        $nodeElseExpr = $nodeElse->expr;
+
+        if (! $nodeIfExpr instanceof LNumber || ! $nodeElseExpr instanceof LNumber) {
+            return null;
+        }
+
+        $node->if = new LNumber($nodeIfExpr->value);
+        $node->else = new LNumber($nodeElseExpr->value);
+
```

Comment:
Does it work only with nubmers? What about strings, floats, basically any other value?

Reply:
Fixed